packages/modules/vehicles/mercedeseq/api.py

```diff
@@ -0,0 +1,209 @@
+        socDebugLog(text)
+
+    elif status_code == 404:
+        socDebugLog(what + " Request fehlgeschlagen Code: " + str(status_code) +
+                    " (The requested resource was not found, e.g.: the selected vehicle could not be found)")
+        socDebugLog(text)
+
+    elif status_code == 429:
+        socDebugLog(what + " Request fehlgeschlagen Code: " + str(status_code) +
+                    " (The service received too many requests in a given amount of time)")
+        socDebugLog(text)
+
+    elif status_code == 500:
+        socDebugLog(what + " Request fehlgeschlagen Code: " + str(status_code) +
+                    " (The service received too many requests in a given amount of time)")
+        socDebugLog(text)
+
+    elif status_code == 503:
+        socDebugLog(what + " Request fehlgeschlagen Code: " + str(status_code) +
+                    " (The server is unable to service the request due to a temporary unavailability condition)")
+        socDebugLog(text)
+
+    else:
+        socDebugLog(what + " Request fehlgeschlagen unbekannter Code: " + str(status_code))
+        socDebugLog(text)
+
+
+def fetch_soc(config: MercedesEQSoc,
+              vehicle_id: int) -> Tuple[float, float]:
+    if Debug >= 1:
+        socDebugLog("Debug Level: " + str(Debug))
+
+    client_id = config.configuration.client_id
+    client_secret = config.configuration.client_secret
+    vin = config.configuration.vin
+    soc_url = soc_url_pre_vin + str(vin) + soc_url_post_vin
+    vehicle = vehicle_id
+    if Debug >= 1:
+        socDebugLog("client: " + client_id)
+
+    if Debug >= 2:
+        socDebugLog("SOC URL: " + soc_url)
+
+    # get Access Token from Broker
+    access_token = config.configuration.token.access_token
+    refresh_token = config.configuration.token.refresh_token
+    expires_in = config.configuration.token.expires_in
+    if Debug >= 2:
+        socDebugLog("Conf Access Tok: " + access_token)
+        socDebugLog("Conf Refresh Tok: " + refresh_token)
+        socDebugLog("Conf Expires_in: " + str(expires_in))
+
+    # socDebugLog("Token expires at: " +time.strftime( "%d.%m.%Y  %H:%M:%S",time.localtime(expires_in)))
+    socDebugLog("Token expires in: " + str(int(expires_in) - int(time.time())) + "s. at: " +
+                time.strftime("%d.%m.%Y  %H:%M:%S", time.localtime(expires_in)))
+
+    if int(expires_in) < int(time.time()):
+        # Access Token is exired
+        if Debug >= 1:
+            socDebugLog("Acc Token Expired")
+
+        # get new Access Token with referesh token
+        data = {'grant_type': 'refresh_token', 'refresh_token': refresh_token}
+        ref = requests.post(tok_url, data=data, verify=True, allow_redirects=False, auth=(client_id, client_secret),
+                            timeout=req_timeout)
+        if Debug >= 1:
+            socDebugLog("Refresh Token Call:" + str(ref.status_code))
+            socDebugLog("Refresh Token Text:" + str(ref.text))
```

Comment:
```suggestion
        ref = req.get_http_session().post(tok_url, data=data, verify=True, allow_redirects=False, auth=(client_id, client_secret),
                            timeout=req_timeout)
```
Das req-Modul unter packages/modules/common/req.py loggt per Hook automatisch Antwort und ggf Fehler.

Reply:
done